libs/experimental/langchain_experimental/language_detector/lang_detect.py

```diff
@@ -0,0 +1,34 @@
+from typing import List, Tuple
+
+from langchain_experimental.language_detector.base import LanguageDetectorBase
+
+try:
+    import langdetect
+except ImportError:
+    raise ImportError(
+        "Could not import langdetect, please install with " "`pip install langdetect`."
+    )
+
+
+class LangDetector(LanguageDetectorBase):
+    """
+    Language detector based on langdetect package.
+    Warning: Results are not deterministic!
+    It supports 55 languages out of the box.
+    """
+
+    def __init__(self, threshold: float = 0.1):
+        self.threshold = (
+            threshold  # Minimum probability to consider a language as detected.
```

Comment:
I don't see much value in threshold here, what if I want to have a streamlit app with slider for it? it is ugly to re-create object to change threshold value. Also the default I expect to see is zero - it should return all detected languages. Also, function at line 25 does not use threshold nor return it - it already is confusing.